lib/db/instance.go

```diff
@@ -297,7 +297,7 @@ func (db *instance) withGlobal(folder, prefix []byte, truncate bool, fn Iterator
 
 		fk = db.keyer.GenerateDeviceFileKey(fk, folder, vl.Versions[0].Device, name)
 
-		f, ok := db.getFileTrunc(fk, truncate)
+		f, ok := t.getFileTrunc(fk, truncate)
```

Comment:
From consistency perspective, shouldn't the root check at the top of this function happen part of the transaction?

Reply:
I considered it, but since it just reads one value it's equivalent anyway and taking the snapshot looked like overhead. However I see now that `getGlobal` actually takes a snapshot of its own, so it would be cleaner to do that... I'll tweak.

Reply:
Same for `withHave`. As far as I understood, taking the snapshot is no or negligible overhead, because `db.Get` internally takes a snapshot too. So first doing `db.Get` then in most cases starting a transaction and thus taking a seconds snapshot seems like more overhead.